src/hooks/useTaskProgress.tsx

```diff
@@ -0,0 +1,53 @@
+import { useMemo, useCallback } from 'react';
+import { Newbie, Task } from 'buddy-app-schema';
+import { isCompleted } from 'utils';
+
+interface TaskProgress {
+  getBuddyProgress: (newbie: Newbie) => number;
+  buddyProgress: number;
+  newbieProgress: number;
+}
+
+const isCompletedTask = (task: Task | null) => task && isCompleted(task.status);
+
+const transformNewbieProgress = (newbie?: Newbie): number => {
```

Comment:
@Tonik1204 can you add some unit tests to this file?

Reply:
Fixed here [#90d27e2](https://github.com/BuildIt-Poland/buddy-app/pull/61/commits/90d27e2ca5c70e562b45c061adbf45419e4df24f)